portal-backend/depmap/private/views.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+from depmap.utilities.sign_bucket_url import sign_url
+from flask import (
+    Blueprint,
+    abort,
+    current_app,
+    redirect,
+)
+from oauth2client.service_account import ServiceAccountCredentials
+
+log = logging.getLogger(__name__)
+
+blueprint = Blueprint(
+    "private", __name__, url_prefix="/private", static_folder="../static"
+)
+
+
+@blueprint.route("/<bucket_name>/<filename>")
+def portal_file_private_url(bucket_name, filename):
+    private_file_buckets = current_app.config.get("PRIVATE_FILE_BUCKETS")
+    # Make sure bucket provided is accepted in valid list of buckets
+    if private_file_buckets is None or bucket_name not in private_file_buckets:
+        abort(404)
+
+    try:
+        credentials = ServiceAccountCredentials.from_json_keyfile_name(
+            current_app.config["DOWNLOADS_KEY"]
+        )
+        signed_url = sign_url(credentials, bucket_name, filename)
+    except Exception as e:
+        raise e
+    # Redirect to generated signed GCS url
+    return redirect(signed_url)
```

Comment:
The current behavior for this is that on clicking this portal link, it automatically starts the download. I would like to open a new tab to view the file/video but there doesn't seem to be a flask function to do so. Any suggestions?

Reply:
Also, this is just my own opinion of how I'd like this endpoint to work

Reply:
This might be easier to talk about. There are some options but let's discuss so I can make sure I'm on the same page.